src/bitset.hpp

```diff
@@ -94,18 +113,35 @@ class Bitset {
   // separated by a "|".
   std::string ToStringChunked(size_t chunk_count) const;
   std::string SubsplitToString() const;
+  std::string SubsplitToNumString() const;
   std::string SubsplitToIndexSetString() const;
-  // Is this the subsplit of a leaf node.
+  // Is this the subsplit of a leaf node?
   bool SubsplitIsLeaf() const;
-  // Is this the subsplit of a rootsplit.
+  // Is this the subsplit of root node?
+  bool SubsplitIsRoot() const;
+  // Is this the subsplit of a rootsplit?
   bool SubsplitIsRootsplit() const;
   // Is this the rotated child of the given subsplit?
   bool SubsplitIsRotatedChildOf(const Bitset &other) const;
   // Is this the sorted child of the given subsplit?
   bool SubsplitIsSortedChildOf(const Bitset &other) const;
   // Get the union of the two chunks.
   Bitset SubsplitChunkUnion() const;
-
+  // Get whether this node is the sorted of rotated child of given subsplit.
+  // Returns -1 given bitset if it is not a parent.
+  int SubsplitWhichChildOf(const Bitset &parent) const;
+  // Check whether bitset represents a valid subsplit.
+  bool SubsplitIsValid() const;
+  // NNI: Nearest Neighbor Interchange.
+  // Given a parent/child pair of subsplits embedded within a subsplit DAG, produces the
+  // resulting parent/child subsplit pair that result from an NNI operation.
+  static BitsetPair SubsplitNNI(const Bitset &parent_in, const Bitset &child_in,
+                                const size_t swap_which_child_with_sister,
+                                const size_t which_child_of_parent);
+  static BitsetPair SubsplitNNI(const Bitset &parent_in, const Bitset &child_in,
```

Comment:
Essentially, if you know whether `child_in` is the right or left child of `parent_in`, you can feed it into the function.  Otherwise the function can infer the relationship from the Bitset.

Reply:
Great! Just add a comment saying what you have here.

Reply:
Added a comment.